File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum, max_sum = nums[0], nums[0]
        for i in range(1, len(nums)):
            num = nums[i]
            cur_sum = max(cur_sum + num, num)
            max_sum = max(max_sum, cur_sum)
        return max_sum

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int largestSum = INT_MIN;
        int currentSum = -1;

        for (auto num: nums) {
            currentSum = max(num, num + currentSum);
            largestSum = max(currentSum, largestSum);
        }
        return largestSum;
    }
};